src/parseExpr_staged.jl

```diff
@@ -655,13 +655,22 @@ function parseExpr(x, aff::Symbol, lcoeffs::Vector, rcoeffs::Vector, newaff::Sym
                 push!(blk.args, :($newaff = $callexpr))
                 return newaff, blk
             end
-        elseif x.head == :call && x.args[1] == :^ && is_complex_expr(x.args[2])
+        elseif x.head == :call && x.args[1] == :^ && is_complex_expr(x.args[2]) && x.args[3] == 2
             blk = Expr(:block)
             s = gensym()
             newaff_, parsed = parseExprToplevel(x.args[2], s)
             push!(blk.args, :($s = 0.0; $parsed))
             push!(blk.args, :($newaff = $aff + $(Expr(:call,:*,lcoeffs...,newaff_,newaff_,rcoeffs...))))
             return newaff, blk
+        elseif x.head == :call && x.args[1] == :^ && is_complex_expr(x.args[2]) && x.args[3] == 1
+            blk = Expr(:block)
+            s = gensym()
+            newaff_, parsed = parseExprToplevel(x.args[2], s)
+            push!(blk.args, :($s = 0.0; $parsed))
+            push!(blk.args, :($newaff = $aff + $(Expr(:call,:*,lcoeffs...,newaff_,rcoeffs...))))
+            return newaff, blk
+        elseif x.head == :call && x.args[1] == :^ && is_complex_expr(x.args[2]) && x.args[3] == 0
```

Comment:
This misses the case of a non-constant exponent and complex subexpression. I think the following code will crash: ``sum(x[i] for i in 1:10)^n``